src/util/std_expr.h

```diff
@@ -3684,39 +3672,34 @@ inline void validate_expr(const array_update_exprt &value)
 
 /*! \brief Extract member of struct or union
 */
-class member_exprt:public exprt
+class member_exprt:public unary_exprt
 {
 public:
-  explicit member_exprt(const exprt &op):exprt(ID_member)
+  explicit member_exprt(const exprt &op):unary_exprt(ID_member, op)
   {
-    copy_to_operands(op);
   }
```

Comment:
I'm wondering whether this constructor should be removed as the `member_exprt` remains half-constructed for it lacks a `component_name`.

Actually, it's worse: this is wrong, as it would yield a `member_exprt` with a struct type rather than the member's type.

Reply:
Indeed, this should ideally go away.